elasticgraph-warehouse/spec/integration/elastic_graph/warehouse/schema_definition/rake_tasks_spec.rb

```diff
@@ -0,0 +1,219 @@
+            EOS
+
+            run_rake_with_warehouse("schema_artifacts:dump")
+
+            warehouse_config = YAML.safe_load(read_artifact(Warehouse::DATA_WAREHOUSE_FILE))
+            expect(warehouse_config["tables"].keys).to eq(%w[apples middles zebras])
+          end
+
+          it "includes all warehouse table configurations" do
+            write_warehouse_schema(table_defs: <<~EOS)
+              s.object_type "Product" do |t|
+                t.field "id", "ID"
+                t.field "name", "String"
+                t.field "price", "Float"
+                t.index "products" do |i|
+                  i.warehouse_table "products"
+                end
+              end
+
+              s.object_type "User" do |t|
+                t.field "id", "ID"
+                t.field "email", "String"
+                t.index "users" do |i|
+                  i.warehouse_table "users"
+                end
+              end
+
+              s.object_type "Order" do |t|
+                t.field "id", "ID"
+                t.index "orders" do |i|
+                  i.warehouse_table "orders"
+                end
+              end
+            EOS
+
+            run_rake_with_warehouse("schema_artifacts:dump")
+
+            warehouse_config = YAML.safe_load(read_artifact(Warehouse::DATA_WAREHOUSE_FILE))
+            expect(warehouse_config["tables"].keys).to contain_exactly("orders", "products", "users")
+
+            warehouse_config["tables"].each do |table_name, table_config|
+              expect(table_config).to have_key("table_schema")
+              expect(table_config["table_schema"]).to start_with("CREATE TABLE IF NOT EXISTS #{table_name}")
+            end
+          end
+        end
+
+        def write_warehouse_schema(table_defs:)
+          ::File.write("schema.rb", <<~EOS)
+            ElasticGraph.define_schema do |s|
+              s.json_schema_version 1
+
+              # Add a dummy indexed type to ensure the Query type has at least one field.
+              # This prevents GraphQL-Ruby warnings about empty Query types in tests.
+              s.object_type "_DummyWarehouseTestType" do |t|
+                t.field "id", "ID"
+                t.index "dummy_warehouse_test_indices"
+              end
+
+              #{table_defs}
+            end
+          EOS
+        end
+
+        def run_rake_with_warehouse(*args, enforce_json_schema_version: true)
+          run_rake(*args) do |output|
+            ElasticGraph::SchemaDefinition::RakeTasks.new(
+              schema_element_name_form: :snake_case,
+              index_document_sizes: false,
+              path_to_schema: "schema.rb",
+              schema_artifacts_directory: "config/schema/artifacts",
+              enforce_json_schema_version: enforce_json_schema_version,
+              extension_modules: [Warehouse::SchemaDefinition::APIExtension],
+              output: output
+            )
+          end
+        end
+
+        def read_artifact(name)
```

Comment:
Every caller of this method passes the exact same `name` argument: `Warehouse::DATA_WAREHOUSE_FILE`.  WDYT about defining this as `read_warehouse_artifact` (with no args) and having it use `DATA_WAREHOUSE_FILE` internally?